openassessment/assessment/test/test_staff.py

```diff
@@ -0,0 +1,348 @@
+        "singing": "no",
+    }
+}
+
+# Answers are against RUBRIC_DICT -- this is worth 14 points
+# Feedback text is one character over the limit.
+LONG_FEEDBACK_TEXT = u"是" * Assessment.MAX_FEEDBACK_SIZE + "."
+ASSESSMENT_DICT_HUGE = {
+    'overall_feedback': LONG_FEEDBACK_TEXT,
+    'criterion_feedback': {
+        "secret": LONG_FEEDBACK_TEXT,
+        u"ⓢⓐⓕⓔ": LONG_FEEDBACK_TEXT,
+        "giveup": LONG_FEEDBACK_TEXT,
+        "singing": LONG_FEEDBACK_TEXT,
+    },
+    'options_selected': {
+        "secret": "yes",
+        u"ⓢⓐⓕⓔ": "yes",
+        "giveup": "eager",
+        "singing": "no",
+    },
+}
+
+
+
+@ddt
+class TestStaffOverwrite(CacheResetTest):
+    """
+    Tests for staff assessments made as overrides, when none is required to exist.
+    """
+
+    STEPS = ['self']
+    STEP_REQUIREMENTS = {}
+    STEP_REQUIREMENTS_WITH_STAFF = {'staff': {'required': True}}
+
+    def test_create_assessment_points(self):
+        # Create assessment
+        tim_sub, tim = self.create_student_and_submission("Tim", "Tim's answer")
+
+        # Staff assess it
+        assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            ASSESSMENT_DICT['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+
+        # Ensure points are calculated properly
+        self.assertEqual(assessment["points_earned"], 6)
+        self.assertEqual(assessment["points_possible"], 14)
+
+        #ensure submission is marked as finished
+        self.assertTrue(staff_api.assessment_is_finished(tim_sub["uuid"], self.STEP_REQUIREMENTS))
+
+    def test_create_assessment_overrides(self):
+        # Create assessment
+        tim_sub, tim = self.create_student_and_submission("Tim", "Tim's answer")
+
+        # Self assess it
+        self_assessment = self_assess(
+            tim_sub["uuid"],
+            tim["student_id"],
+            ASSESSMENT_DICT['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+        # and update workflow with new scores
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+
+        # Verify both assessment and workflow report correct score
+        self.assertEqual(self_assessment["points_earned"], 6)
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], 6)
+
+        # Now override with a staff assessment
+        staff_assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            ASSESSMENT_DICT_PASS['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+        # Be sure to update the workflow!
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS, force_update_score=True)
+
+        # Verify both assessment and workflow report correct score
+        self.assertEqual(staff_assessment["points_earned"], 14)
```

Comment:
It would be good to have scores that weren't the maximum to be sure that they are being derived correctly.